client/mysqlbinlog.cc

```diff
@@ -955,14 +955,14 @@ static bool print_row_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
   }
 
   /* skip the event check */
-  if (skip_event)
+  if (skip_event && !(opt_flashback && is_stmt_end))
```

Comment:
What #1674 did was copy-pasting the `else` branch of the next `if` to the previous `if` tree as a new branch for `(skip_event && opt_flashback && is_stmt_end)`.
Here, I reimplement it by skipping this intermediate “step” (?) for that exact condition.

While the diff is now smaller, it has become proof that our code is 🍝.